tests/PHPStan/Analyser/nsrt/isset-coalesce-empty-type.php

```diff
@@ -472,7 +472,7 @@ function coalesce()
 
 		assertType('int<0, max>', rand() ?? false);
 
-		assertType('0|lowercase-string', preg_replace('', '', '') ?? 0);
+		assertType('0|(lowercase-string&uppercase-string)', preg_replace('', '', '') ?? 0);
```

Comment:
This type shouldn't be possible. My guess is that it wouldn't survive type normalization. Because from their implementation of isSuperTypeOf, these types are mutually exclusive. For the same reason `string&int` can't be created.

I understand it's possible for a string to comply with both constraints, but I'd prefer if this extension produced just a general string instead of `lowercase-string&uppercase-string`.

Reply:
Losing `lowercase-string&uppercase-string` would create issue, even for users of only one type

```
/** @param lowercase-string $string */
public function acceptsOnlyLowercase($string) {}

/** @var int $int */
$int = ...;

acceptsOnlyLowercase((string) $int) {} // Will report an error
```

`(string) $int` is both a lowercase and uppercase string, so should be accepted by `acceptsOnlyLowercase` but if we lose those accessory in normalization ; wouldn't we get an error "lowercase-string expected, string passed" ?

Reply:
I think @VincentLanglet is right -- strtoupper('11') and strtolower('11') both result in '11', which means '11' is both uppercase and lowercase. Am I missing something?

Reply:
ondrej seems to be on the same page, see https://github.com/phpstan/phpstan-src/pull/3613#issuecomment-2469975969